File: AnalyseActivityLifeCycle-master/app/src/main/java/com/example/analyseactivitylifecycle/MainActivity.java
package com.example.analyseactivitylifecycle;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private final String Tag = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d(Tag, "onCreate");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(Tag, "onStart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(Tag, "onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(Tag, "onPause");

    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(Tag, "onStop");

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(Tag, "onDestroy");

    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d(Tag, "onRestart");
    }
}

File: Kotlin Projects/CoroutinesWE/app/src/main/java/com/example/coroutines_we/MainActivity.kt
package com.example.coroutines_we

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        runBlocking {
            CoroutineScope(Dispatchers.IO).launch {
                Log.d("Coroutines", "launch")
                val  data = getDataFromApi()
                Log.d("Coroutines", "First Coroutines $data")

                val wait  = async {
                    val data = getDataFromApi()
                    return@async data
                }.await()

                val res2  = async {
                    val data2 = getDataFromApi()
                    return@async data2
                }.await()

                CoroutineScope(Dispatchers.Main).launch {
                    tvTextView.text = "$data + $res2"
                    Log.d("Coroutines", "Second Coroutines $wait + $res2")
                }

            }
        }
    }

    suspend fun getDataFromApi(): String {
        delay(6000);
        return "Data fetch Successfully"
    }
}


/*
            val wait  = async {
                val data = getDataFromApi()
                return@async data
            }.await()

 */